opm/grid/common/GridPartitioning.cpp

```diff
@@ -468,123 +549,132 @@ void addOverlapLayer(const CpGrid& grid, int index, const CpGrid::Codim<0>::Enti
+    {
+        buffer->resize(proc.second);
+        MPI_Irecv(buffer->data(), proc.second, MPI_INT, proc.first, tag, cc, &(*req));
+        ++req; ++buffer;
+    }
 
-        for(const auto& proc: exportProcs)
-        {
-            std::vector<int> sendBuffer;
-            sendBuffer.reserve(proc.second);
-            std::for_each(ownerEnd, exportList.end(),
-                          [&sendBuffer, &proc](const std::tuple<int,int,char>& t)
-                          {
-                              if ( std::get<1>(t) == proc.first )
-                                  sendBuffer.push_back(std::get<0>(t));
-                          });
-            MPI_Send(sendBuffer.data(), proc.second, MPI_INT, proc.first, tag, cc);
-        }
+    for(const auto& proc: exportProcs)
+    {
+        std::vector<int> sendBuffer;
+        sendBuffer.reserve(proc.second);
+        std::for_each(ownerEnd, exportList.end(),
+                      [&sendBuffer, &proc](const std::tuple<int,int,char>& t)
+                      {
+                          if ( std::get<1>(t) == proc.first )
+                              sendBuffer.push_back(std::get<0>(t));
+                      });
+        MPI_Send(sendBuffer.data(), proc.second, MPI_INT, proc.first, tag, cc);
+    }
 
-        std::inplace_merge(exportList.begin(), ownerEnd, exportList.end());
+    std::inplace_merge(exportList.begin(), ownerEnd, exportList.end());
 
-        MPI_Waitall(requests.size(), requests.data(), statuses.data());
-        buffer = receiveBuffers.begin();
-        auto importOwnerSize = importList.size();
+    MPI_Waitall(requests.size(), requests.data(), statuses.data());
+    buffer = receiveBuffers.begin();
+    auto importOwnerSize = importList.size();
 
-        for(const auto& proc: importProcs)
-        {
-            for(const auto& index: *buffer)
-                importList.emplace_back(index, proc.first, AttributeSet::copy, -1);
-            ++buffer;
-        }
-        std::sort(importList.begin() + importOwnerSize, importList.end(),
-                  [](const std::tuple<int,int,char,int>& t1, const std::tuple<int,int,char,int>& t2)
-                  { return std::get<0>(t1) < std::get<0>(t2);});
-        return importOwnerSize;
+    for(const auto& proc: importProcs)
+    {
+        for(const auto& index: *buffer)
+            importList.emplace_back(index, proc.first, AttributeSet::copy, -1);
+        ++buffer;
+    }
+    std::sort(importList.begin() + importOwnerSize, importList.end(),
+              [](const std::tuple<int,int,char,int>& t1, const std::tuple<int,int,char,int>& t2)
+              { return std::get<0>(t1) < std::get<0>(t2);});
+    return importOwnerSize;
 #else
-        (void) grid;
-        (void) cell_part;
-        (void) exportList;
-        (void) importList;
-        (void) cc;
-        (void) layers;
-        DUNE_THROW(InvalidStateException, "MPI is missing from the system");
-
-        return 0;
+    static_cast<void> grid;
```

Comment:
this is not valid code and breaks the sequential build. furthermore, please use [[maybe_unused]] instead of these messy casts.

Reply:
Sorry, fixing it in OPM/opm-grid#848